openstack_controller/datadog_checks/openstack_controller/api.py

```diff
@@ -101,15 +105,45 @@ def get_project_limits(self, tenant_id):
         limits = server_stats.get('limits', {}).get('absolute', {})
         return limits
 
-    def get_flavors_detail(self, query_params, timeout=None):
+    def get_flavors_detail(self, query_params):
         url = '{}/flavors/detail'.format(self.endpoint)
-        flavors = self._make_request(url, self.headers, params=query_params, timeout=timeout)
-        return flavors.get('flavors', [])
+        return self._get_paginated_list(url, 'flavors', query_params)
+
+    def _get_paginated_list(self, url, obj, query_params):
+        result = []
+        query_params = query_params or {}
+        query_params['limit'] = self.paginated_limit
+        resp = self._make_request(url, self.headers, params=query_params)
+        result.extend(resp.get(obj, []))
+        # Avoid the extra request since we know we're done when the response has anywhere between
+        # 0 and paginated_server_limit servers
+        while len(resp) == self.paginated_limit:
+            query_params['marker'] = resp[-1]['id']
+            query_params['limit'] = self.paginated_limit
+            retry = 0
+            while retry < DEFAULT_MAX_RETRY:
+                # `details` endpoints are typically expensive calls, if it fails,
```

Comment:
looks like we can remove "if it fails" at the end of this line as its duplicated on the next.